samples/AugmentedInferenceAgreesWithBaseInference.java

```diff
@@ -24,16 +24,21 @@ void x(Foo<Object> a, Foo<@Nullable Object> b) {
     List<Foo<?>> l1 = makeList(a, b);
 
     /*
-     * List of some unspecified homogeneous Foo type. There is such a type under plain Java (since
-     * the base type for both is Foo<Object>) but not under JSpecify (since one is Foo<Object> and
-     * the other is Foo<@Nullable Object>).
+     * List of some unspecified (possibly homogeneous) Foo type. javac infers `makeList` to have the
+     * type List<Foo<Object>>, but that's a problem for JSpecify, which needs List<Foo<?>>.
      *
      * Notice that `makeList(a, b)` is fine "in a vacuum" even under JSpecify (as shown above). Only
-     * here, where the type of the expression is forced to conform to the target type, is there a
-     * problem.
+     * here, where the type of the expression is forced to conform to the javac's type for the
```

Comment:
It looks like the expression type is being forced to conform to _JSpecify's_ preferred type. No?